python/Ganga/Runtime/GPIexport.py

```diff
@@ -18,6 +18,31 @@
 from Ganga.GPIDev.Base.Objects import GangaObject
 from inspect import isclass
 
+def _addToInterface(interface, name, _object):
+
+    if isType(_object, GangaObject):
+        setattr(interface, name, addProxy(_object))
+    elif isclass(_object) and issubclass(_object, GangaObject):
+        setattr(interface, name, getProxyClass(_object))
+    else:
+        setattr(interface, name, _object)
+
+def exportToInterface(myInterface, name, _object, doc_section, docstring=None):
```

Comment:
I understand the need for using a `RegistrySlice` in some contexts. I just think that getting it via the GPI is not the best way to do it. This is somewhere we need a case-by-case process to check them. For example, I'm almost certain the monitoring could use a plain dictionary for `JobRegistry_Monitor.registry`.

As for looking up `"LocalFile"`. In what situations do we need to do this without any context? Usually we're loading a job (or similar) from disk and find a `class="LocalFile"` in which case we know what plugin category to look in since we know what category there is on that schema item (because it's a `ComponentItem`).

In the situation where we're given a contextless `"LocalFile"` (I wonder where this happens?) then in reality there's never a plugin objecte in more than one category with the same name (since that would break when exporting to the GPI) in which case adding a function to `PluginManager` to find a name by searching all categories would be a good idea.

I really dislike loading a namespace and then searching it's module `__dict__` for a match like we do in a few places at the moment.